### PR TITLE
Event lists bug fixes and improvements

### PR DESCRIPTION
**Pull request by Cell00phane**
- Will display no upcoming events if no events in event list fragment
- document, clean code and remove todos
- Delete all events on logout
- Do not delete followed events on user login
- user settings related to location are on by default

### CODE REVIEW

File: app/src/main/res/layout/fragment_events.xml [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
+    <TextView
+        android:id="@+id/fragment_events_no_upcoming_events_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/no_upcoming_events"
+        android:visibility="gone"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
```
Comment (AlessioVerardo) [inline app/src/main/res/layout/fragment_events.xml:30]: Nice improvement !

File: app/src/main/java/com/github/sdpteam15/polyevents/model/database/local/entity/UserSettings.kt [hunk 2]
```diff
@@ -26,9 +26,9 @@ data class UserSettings(
     // Ideally should have its setter private but Room needs it to reconstruct objects
     val userUid: String = DEFAULT_ID,
     @ColumnInfo(name = "track_location")
-    var trackLocation: Boolean = false,
+    var trackLocation: Boolean = true,
     @ColumnInfo(name = "is_sending_location_on")
-    var isSendingLocationOn: Boolean = false,
+    var isSendingLocationOn: Boolean = true,
```
Comment (AlessioVerardo) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/database/local/entity/UserSettings.kt:31]: Cool improvement

Comment (Cell00phane) [reply app/src/main/java/com/github/sdpteam15/polyevents/model/database/local/entity/UserSettings.kt:31]: Thank you

**Review by AlessioVerardo — APPROVED** [review]
Nice refactor with a lot of comments ! Great job

File: app/src/androidTest/java/com/github/sdpteam15/polyevents/view/fragments/EventListFragmentTest.kt [hunk 3]
```diff
@@ -141,6 +146,21 @@ class EventListFragmentTest {
...
+        FakeDatabaseEvent.events.clear()
+
+        fragmentScenario.recreate()
+
+        assertDisplayed(
+            R.id.fragment_events_no_upcoming_events_text_view,
+            R.string.no_upcoming_events
+        )
+    }
+
```
Comment (Feez2403) [inline app/src/androidTest/java/com/github/sdpteam15/polyevents/view/fragments/EventListFragmentTest.kt:163]: great use of Fakedatabase ! :)

**Review by Feez2403 — APPROVED** [review]
Nice comments

**Review by Rygone — APPROVED** [review]

**Pull request merged**